### DISCUSSION TITLE
[Bug]: TypeError Promise.withResolvers is not a function

### DISCUSSION THREAD

**Discussion by techpranav** [mozilla/pdf.js#18821] [Q&A] [answered]
### Attach (recommended) or Link to PDF file

![image](https://github.com/user-attachments/assets/4df99ab0-9da2-43e3-ab95-158fce405909)
![image](https://github.com/user-attachments/assets/810eb70a-1ab5-44f5-b6a1-9f9ad0b4c1cc)

### Web browser and its version

Electron JS Application

### Operating system and its version

Windows 11

### PDF.js version

4.6.82

### Is the bug present in the latest PDF.js version?

Yes

### Is a browser extension

No

### Steps to reproduce the problem

1.Error occurs when opening PDF in electron app

### What is the expected behavior?

Open PDF in read mode

### What went wrong?

pdf.mjs:11172 Uncaught (in promise) TypeError: Promise.withResolvers is not a function
    at new PDFDocumentLoadingTask (pdf.mjs:11172:1)
    at Module.getDocument (pdf.mjs:10985:1)
    at loadPdf (PDFViewer.js:33:27)
    at PDFViewer.js:39:5
    at commitHookEffectListMount (react-dom.development.js:23189:1)
    at commitPassiveMountOnFiber (react-dom.development.js:24965:1)
    at commitPassiveMountEffects_complete (react-dom.development.js:24930:1)
    at commitPassiveMountEffects_begin (react-dom.development.js:24917:1)
    at commitPassiveMountEffects (react-dom.development.js:24905:1)
    at flushPassiveEffectsImpl (react-dom.development.js:27078:1)

**Comment by Tej-Sharma** [comment]
@techpranav Were you able to figure it out?

**Reply by techpranav** [reply]
No not yet

**Reply by Tej-Sharma** [reply]
I used react-pdf-viewer if you were looking for something for electron

**Comment by midhuhu** [comment]
Has there been any progress on this issue?

**Comment by Snuffleupagus** [comment, marked as answer]
Please see https://github.com/mozilla/pdf.js/wiki/Frequently-Asked-Questions#faq-support, which will tell you two things:
 - We don't support Electron, so you're "on your own" there unfortunately.
 - For older browsers, use a `legacy`-build of the PDF.js library.

**Reply by DenysVuika** [reply]
It's failing with Firefox 137.0

**Comment by Tej-Sharma** [comment]
Oh okay, I ended up using react-pdf-viewer and purchased the license there if anyone's using this for React + Electron and it worked there (interestingly, it's built on react-pdf which uses the worker from pdf.js 🤔)

**Reply by Snuffleupagus** [reply]
> Oh okay, I ended up using react-pdf-viewer and purchased the license there if anyone's using

*A word of warning:* Looking at https://github.com/react-pdf-viewer/react-pdf-viewer/issues there are numerous issues that *appear* to suggest that that project may no longer being actively maintained/supported.

**Comment by vicapow** [comment]
just some perspective, I've also had to polyfill this manually in the source code of the worker as a workaround for users who were reporting issues with this. I think there version of chrome was a little outdated by not that much. I see the `withResolvers` was only added just under 1 year ago. What's the oldest browser version you consider "modern"? 

<img width="801" alt="Screenshot 2024-10-30 at 3 48 26 PM" src="https://github.com/user-attachments/assets/e35dcca0-7f20-47df-9eaf-36d1bd557570">

**Reply by Hellomik2002** [reply]
Can you your pollyfills please ?
